clarification to the MultiAppInterpolationTransfer documentation (#17927)
idaholab/moose: framework/doc/content/source/transfers/MultiAppInterpolationTransfer.md

# MultiAppInterpolationTransfer

> The `MultiAppInterpolationTransfer` transfers the source variable to the nearest node on the

Reviewer: `the nearest nodes`

target mesh using mesh interpolation, including the ability to utilize the displaced
configuration for either or both the source and target.  

> Nodal transfers using the default settings for this interpolation scheme, `interp_type=inverse_distance` and `num_points=3`, are 

Reviewer: I am not sure I am convinced by "num_points=3". I could imagine it is problem dependent  ...

Reviewer 2: He's just saying `num_points=3` because that is the default

Reviewer: Oh, I see. What I meant was that it was hard to say the default setting work best (`best suited` ) for something.

Author: I changed the wording around this.

best suited for the interpolation of point cloud data in the source mesh onto a target mesh.  
The [MultiAppMeshFunctionTransfer](/MultiAppMeshFunctionTransfer.md) may be a better choice for nodal transfers 

> between two meshes because the element shape functions will be used in the transfer.

Reviewer: MultiAppMeshFunctionTransfer does not offer extrapolation, and MultiAppInterpolationTranser offers  extrapolation. That is, a target point value can be computed using the nearest source nodes when a target point is not inside of the source domain.

Reviewer 3: @lynnmunday can you move this to a "see also" section and address Fande's comment?

Reviewer 3: actually you mention the non-extrapolation nature above. this is a bit disjointed

Using `MultiAppInterpolationTranser` with the default interpolation settings for mesh to mesh nodal transfers of data 
can lead to a parallel inconsistency that arises when the point being interpolated to on the
target mesh is exactly the same distance away from more than `num_points` on the source mesh.  

This node-to-node transfer inconsistency could occur on a structured mesh if the target mesh is a refined version of the source mesh.  
In this scenario, a target node will be placed equidistant from several source nodes.  
For a 2D structured mesh of quadrilaterals, the refined mesh's target node would be equidistant from 4 nodes on the source mesh.  
With the default `num_points=3`, the three points chosen from the four equidistant nodes on the source mesh will be arbitrary and has been shown to be dependent on the parallel decomposition of the mesh.  
This inconsistency goes away for this case by increasing `num_points=4` so that all of the nodes in the element are used for the interpolation.  

> For unstructured meshes, this parallel inconsistency *should not* be an issue. 

Reviewer: Parallel inconsistency will happen if a target point has the same different distance to different groups of source points, regardless of meshes

Author: I changed all of the "Parallel inconsistency" to "not deterministic"

> However, this transfer should be used with caution even with an unstructured mesh if the mesh becomes more structured near a boundary or if the target mesh is a refined version of the source mesh.

Reviewer: Different users will have different definitions of "transfer".  Parallel inconsistency does not mean we are doing things wrong. It simply means that there are multiple equally  good or equally bad solutions for a given problem. We always can increase the number of interpolation points to resolve this issue.

So I do not think we should warn users there. What we should do is to explain well to users how an interpolation transfer works, and what they would expect.

NearestNodeTransfer share the same feature as well...

Author: I added a better explanation of the interpolation transfer and linked to wikipedia.

# MultiAppInterpolationTransfer

The `MultiAppInterpolationTransfer` transfers the nearest node's source variables to the nearest node on the
target mesh using mesh interpolation, including the ability to utilize the displaced
configuration for either or both the source and target.  
The `MultiAppInterpolationTransfer` also offers extrapolation on non-overlapping domains where the target data 
can be computed from source nodes not inside the target mesh. 

> Other interpolation transfers like the [MultiAppMeshFunctionTransfer](/MultiAppMeshFunctionTransfer.md) are not able to extrapolate data will only work for over-lapping domains. 

Reviewer 3: ```suggestion
Other interpolation transfers like the [MultiAppMeshFunctionTransfer](/MultiAppMeshFunctionTransfer.md) are not able to extrapolate data will only work for overlapping domains. 
```

Nodal transfers using the default settings for this interpolation scheme, `interp_type=inverse_distance` 
and `num_points=3`, will find the three closest points on the source mesh to a node on the target mesh.  
The source data from the three closest nodes will then be interpolated to the target node using [inverse distance weighting](https://en.wikipedia.org/wiki/Inverse_distance_weighting).   
Inverse distance interpolation is best suited for the interpolation of point cloud data in the source mesh onto a target mesh.  
The [MultiAppMeshFunctionTransfer](/MultiAppMeshFunctionTransfer.md) may be a better choice for nodal transfers 
between two meshes because the element shape functions will be used in the transfer.

> Using `MultiAppInterpolationTranser` with the default interpolation settings for mesh to mesh nodal transfers of data 

Reviewer 3: ```suggestion
Using `MultiAppInterpolationTransfer` with the default interpolation settings for mesh to mesh nodal transfers of data 
```

> Using `MultiAppInterpolationTranser` with the default interpolation settings for mesh to mesh nodal transfers of data 

Author: fixed

can be computed from source nodes not inside the target mesh. 
Other interpolation transfers like the [MultiAppMeshFunctionTransfer](/MultiAppMeshFunctionTransfer.md) are not able to extrapolate data and will only work for fully overlapping domains. 

Nodal transfers using the default settings for this interpolation scheme, `interp_type=inverse_distance` 
and `num_points=3`, will find the three closest points on the source mesh to a node on the target mesh.  
The source data from the three closest nodes will then be interpolated to the target node using [inverse distance weighting](https://en.wikipedia.org/wiki/Inverse_distance_weighting).   
Inverse distance interpolation is best suited for the interpolation of point cloud data in the source mesh onto a target mesh.  
The [MultiAppMeshFunctionTransfer](/MultiAppMeshFunctionTransfer.md) may be a better choice for nodal transfers 
between two meshes with fully overlapping domains because the element shape functions will be used in the transfer. 
However, as mentioned above, only the `MultiAppInterpolationTransfer` can be used to extrapolate data between meshes on domains that do not fully overlap. 

> 

Reviewer 3: I would make that a note box

The MultiAppInterpolationTransfer transfers the source variable to the nearest node on the
target mesh using mesh interpolation, including the ability to utilize the displaced
configuration for either or both the source and target.  

> Nodal transfers using this interpolation scheme is best suited for the interpolation of point cloud data in the source mesh onto a target mesh.  

Reviewer 4: ```suggestion
Nodal transfers using this interpolation scheme are best suited for the interpolation of point cloud data in the source mesh onto a target mesh.  
```

> The [MultiAppMeshFunctionTransfer](/MultiAppMeshFunctionTransfer.md) may be a better choice for nodal transfers between two meshes because element data will be used in the transfer and the transfer is not dependent on the number of points used in the transfer.  

Reviewer 4: I think something is wrong with the documentation of this one too actually. Mentions postprocessors when it s using only variables?

Reviewer 4: (does not have to be fixed here)

Author: you are right.  That transfers documentation is really confusing and I think wrong

> Using it for mesh to mesh nodal transfers of data can lead to a parallel inconsistency that arises when the point being interpolated to on the

Reviewer 4: What is it referring to?
I imagine InterpolationTransfer but would be helpful to repeat to avoid any confusion

target mesh is exactly the same distance away from more than `num_points` on the source mesh.  
This inconsistency goes away by increasing `num_points`.   

> For unstructured meshes, this parallel inconsistency should not be an issue.  

Reviewer 4: I d remove this line. It could still happen. And a lot of UM are a little structured near boundaries

Author: I didn't want to freak people out.  I think this transfer usually works.  I can make "should" itallic.
